components/layout_2020/table/layout.rs

```diff
@@ -1921,40 +1921,34 @@ fn get_size_percentage_contribution_from_style(
     }
 }
 
-fn get_sizes_from_style(
+fn get_outer_sizes_from_style(
     style: &Arc<ComputedValues>,
     writing_mode: WritingMode,
+    padding_border_sums: &LogicalVec2<Au>,
 ) -> (LogicalVec2<Au>, LogicalVec2<Au>, LogicalVec2<Au>) {
-    let get_max_size_for_axis = |size: Option<&ComputedLengthPercentage>| {
-        size.and_then(|length_percentage| length_percentage.to_length())
-            .map_or(MAX_AU, Au::from)
+    let box_sizing = style.get_position().box_sizing;
+    let outer_size = |size: LogicalVec2<Au>| match box_sizing {
+        BoxSizing::ContentBox => &size + padding_border_sums,
+        BoxSizing::BorderBox => LogicalVec2 {
+            inline: size.inline.max(padding_border_sums.inline),
+            block: size.block.max(padding_border_sums.block),
+        },
     };
-
-    let max_size = style.max_box_size(writing_mode);
-    let max_size = LogicalVec2 {
-        inline: get_max_size_for_axis(max_size.inline),
-        block: get_max_size_for_axis(max_size.block),
-    };
-
-    let get_size_for_axis = |size: LengthPercentageOrAuto<'_>| {
+    let get_size_for_axis = |size: &LengthPercentageOrAuto<'_>| {
         size.non_auto()
             .and_then(|size| size.to_length())
             .map_or_else(Au::zero, Au::from)
     };
-
-    let min_size = style.min_box_size(writing_mode);
-    let min_size = LogicalVec2 {
-        inline: get_size_for_axis(min_size.inline),
-        block: get_size_for_axis(min_size.block),
-    };
-
-    let size = style.box_size(writing_mode);
-    let size = LogicalVec2 {
-        inline: get_size_for_axis(size.inline),
-        block: get_size_for_axis(size.block),
+    let get_max_size_for_axis = |size: &Option<&ComputedLengthPercentage>| {
```

Comment:
And `get_max_size` here?

Reply:
Oh, of course. Please disregard my comment.

Reply:
OK went back to "for_axis".